core/commands/dht.go

```diff
@@ -227,6 +231,159 @@ var findProvidersDhtCmd = &cmds.Command{
 	Type: notif.QueryEvent{},
 }
 
+var provideRefDhtCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Announce to the network that you are providing given values.",
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("key", true, true, "The key to find providers for.").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.BoolOption("verbose", "v", "Print extra information.").Default(false),
+		cmds.BoolOption("recursive", "r", "Recursively provide entire graph.").Default(false),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		if n.Routing == nil {
+			res.SetError(errNotOnline, cmds.ErrNormal)
+			return
+		}
+
+		rec, _, _ := req.Option("recursive").Bool()
+
+		var keys []key.Key
+		for _, arg := range req.Arguments() {
+			k := key.B58KeyDecode(arg)
+			if k == "" {
+				res.SetError(fmt.Errorf("incorrectly formatted key: ", arg), cmds.ErrNormal)
+				return
+			}
+
+			has, err := n.Blockstore.Has(k)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+
+			if !has {
+				res.SetError(fmt.Errorf("block %s not found locally, cannot provide", k), cmds.ErrNormal)
+				return
+			}
+
+			keys = append(keys, k)
+		}
```

Comment:
What I mean is it will buffer whole stdin, instead of starting process for each line as we go.

but if we don't do it then we hit problem what to do if command errors out on input later.

Ok, let's leave it as it is, and create discussion issue about that.